### PR TITLE
Implement `bal tool search`

### PR DESCRIPTION
**Pull request by gayaldassanayake**
## Purpose
$ subject

Related to https://github.com/ballerina-platform/ballerina-lang/issues/40264

The output is identical to the bal search command output except for the newly added `ID` field.

```
Ballerina Central
=================

|ID                            |PACKAGE                        |DESCRIPTION                    |DATE            |VERSION                        |
|------------------------------|-------------------------------|-------------------------------|----------------|-------------------------------|
|gayals                        |gayaldassanayake/tool_gayals   |1. Run `bal pack` with the ... |2023-06-14-Wed  |2.2.7                          |


1 tools found.
```

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/ToolCommand.java [hunk 1]
```diff
@@ -517,46 +521,46 @@ private boolean deleteSpecificToolVersionCache(String org, String name, String v
...
+            Settings settings;
+            try {
+                settings = RepoUtils.readSettings();
+                // Ignore Settings.toml diagnostics in the search command
+            } catch (SettingsTomlException e) {
+                // Ignore 'Settings.toml' parsing errors and return empty Settings object
+                settings = Settings.from();
+            }
+            CentralAPIClient client = new CentralAPIClient(RepoUtils.getRemoteRepoURL(),
+                    initializeProxy(settings.getProxy()),
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/ToolCommand.java:540]: @Thevakumar-Luheerathan is there anything more for the proxy implementation?

Comment (gayaldassanayake) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/ToolCommand.java]: @Thevakumar-Luheerathan please check if it's correct now.

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/ToolCommand.java [hunk 2]
```diff
@@ -517,46 +521,46 @@ private boolean deleteSpecificToolVersionCache(String org, String name, String v
...
+
+            List<Tool> tools = toolSearchResult.getTools();
+            if (tools != null && tools.size() > 0) {
+                printTools(toolSearchResult.getTools(), RepoUtils.getTerminalWidth());
+            } else {
+                outStream.println("no tools found.");
+            }
+        } catch (CentralClientException e) {
+            String errorMessage = e.getMessage();
+            if (null != errorMessage && !"".equals(errorMessage.trim())) {
```
[the remark this answers is not in the document]
Comment (gayaldassanayake) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/ToolCommand.java:554]: Added a generic error

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/ToolCommand.java (hunk 1 above)
Comment (Thevakumar-Luheerathan) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/ToolCommand.java]: ya... It's

**Review by azinneera — APPROVED** [review]

**Pull request merged**